src/main/java/com/ft/methodecontentcollectionmapper/messaging/MessageBuilder.java

```diff
@@ -1,77 +1,86 @@
-
-            msg.addCustomMessageHeader(TRANSACTION_ID_HEADER, storyPackage.getPublishReference());
-            msg = KeyedMessage.forMessageAndKey(msg, storyPackage.getUuid());
-        } catch (JsonProcessingException e) {
-            throw new StoryPackageMapperException("unable to write JSON for message", e);
-        }
-
-        return msg;
-    }
-
-    public static class MessageBody {
-        @JsonProperty("payload")
-        public final StoryPackage payload;
-        @JsonProperty("contentUri")
-        public final String contentUri;
-        @JsonProperty("lastModified")
-        public final String lastModified;
-        @JsonProperty("uuid")
-        public final String uuid;
-
-        MessageBody(@JsonProperty("payload") StoryPackage payload, @JsonProperty("contentUri") String contentUri,
-                @JsonProperty("lastModified") String lastModified, @JsonProperty("uuid") String uuid) {
-            this.contentUri = contentUri;
-            this.payload = payload;
-            this.lastModified = lastModified;
-            this.uuid = uuid;
-        }
-    }
-
-}
+package com.ft.methodecontentcollectionmapper.messaging;
+
+import static com.ft.api.util.transactionid.TransactionIdUtils.TRANSACTION_ID_HEADER;
+import static java.time.ZoneOffset.UTC;
+
+import com.ft.methodecontentcollectionmapper.model.ContentCollection;
+import com.ft.methodecontentcollectionmapper.model.ContentCollectionType;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.ResolverStyle;
+import java.util.Date;
+import java.util.UUID;
+
+import javax.ws.rs.core.UriBuilder;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.ft.messagequeueproducer.model.KeyedMessage;
+import com.ft.messaging.standards.message.v1.Message;
+import com.ft.messaging.standards.message.v1.SystemId;
+import com.ft.methodecontentcollectionmapper.exception.ContentCollectionMapperException;
+
+public class MessageBuilder {
+    private static final String CMS_CONTENT_PUBLISHED = "cms-content-published";
+    private static final DateTimeFormatter RFC3339_FMT = DateTimeFormatter.ISO_OFFSET_DATE_TIME
+            .withResolverStyle(ResolverStyle.STRICT);
+
+    private final String contentUriPrefix;
+    private final SystemId systemId;
+    private final ObjectMapper objectMapper;
+
+    public MessageBuilder(String contentUriPrefix, String systemId, ObjectMapper objectMapper) {
+        this.contentUriPrefix = contentUriPrefix;
+        this.systemId = SystemId.systemIdFromCode(systemId);
+        this.objectMapper = objectMapper;
+    }
+
+    public Message buildMessage(ContentCollection contentCollection) {
+        String specificContentUriPrefix = contentUriPrefix;
+        if (contentCollection.getType() == ContentCollectionType.STORY_PACKAGE_TYPE) {
```

Comment:
In `EomFileToContentCollectionMapper` you are using a map to convert the from `webType` to `ContentCollectionType` while here you are using a series of if-else statements to convert from `ContentCollectionType` to `specificContentUriPrefix`

I would actually move both of these conversion in the enum:
1) create a static `valueOf` method that converts a string (the `webType`) to a enum value
2) create a abstract method `toContentUriPrefix` and implemented in each enum, so that it can convert itself 

This way, in the future, in case we add an extra type, all changes can be done in the enum file, without changing the mapper or the message builder.